core/services/ocr2/plugins/ocr2keeper/evm21/registry_check_pipeline.go

```diff
@@ -153,7 +143,7 @@ func (r *EvmRegistry) verifyLogExists(upkeepId *big.Int, p ocr2keepers.UpkeepPay
 		r.lggr.Debugf("log block not provided, querying eth client for tx hash %s for upkeepId %s", hexutil.Encode(p.Trigger.LogTriggerExtension.TxHash[:]), upkeepId)
 	}
 	// query eth client as a fallback
-	bn, _, err := r.getTxBlock(p.Trigger.LogTriggerExtension.TxHash)
+	bn, _, err := core.GetTxBlock(r.client, p.Trigger.LogTriggerExtension.TxHash)
```

Comment:
unrelated q. I see in line 149 we look for error `missing required field` which implies `UpkeepFailureReasonTxHashNoLongerExists`. Curious, what is the scenario where we expect the `missing required field` error?

Reply:
[this](https://github.com/smartcontractkit/chainlink/blob/develop/core/chains/evm/client/client.go#L183-L192) makes me think we can check for this string to be safe.

Reply:
hmm interesting. could you add the comment in our code then
`the error that can arise when we're talking to a Parity node that has no receipt yet.`